src/OrchardCore/OrchardCore.OpenId.Core/Services/Managers/OpenIdApplicationManager.cs

```diff
@@ -176,6 +177,25 @@ public virtual ValueTask<string> GetPhysicalIdAsync(TApplication application, Ca
             await base.PopulateAsync(descriptor, application, cancellationToken);
         }
 
+        public override async Task<ImmutableArray<ValidationResult>> ValidateAsync(
+            TApplication application, CancellationToken cancellationToken = default)
+        {
+            var results = ImmutableArray.CreateBuilder<ValidationResult>();
+            results.AddRange(await base.ValidateAsync(application, cancellationToken));
+
+            foreach (var role in await Store.GetRolesAsync(application, cancellationToken))
+            {
+                if (string.IsNullOrEmpty(role))
+                {
+                    results.Add(new ValidationResult("Roles cannot be null or empty."));
```

Comment:
Should it be translated?

Reply:
We could do that, but the validation errors added by OpenIddict itself are not localized (it's something I'd like to do in a future version) so it would be weird to have localized and non-localized strings, no?

Reply:
Also, we don't catch the validation exception to render these messages as part of the UI when validation fails, so it's just a check that only aims at making debugging easier.

Reply:
Exceptions are exceptions, if they are expected they should be caught and a translated message displayed, while we log the actual error. If an exception message is displayed on the UI through some catch-all mechanism, then that's ok.

Validation messages are expected things to happen that are displayed as part of the rendered screen.

It's fine right now, but we should at least create an issue to track it, and maybe someone will do it.